spec.md

```diff
@@ -202,11 +202,13 @@ help intermediate gateways determine how to route the events.
 
 ### source
 * Type: `URI-reference`
-* Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization
-  publishing the event, the process that produced the event, and some unique
-  identifiers. The exact syntax and semantics behind the data encoded in the URI
-  is event producer defined.
+* Description: Identifies the event producer.
+  The source MAY include a URI authority
+  [RFC3986  §3.2](https://tools.ietf.org/html/rfc3986#section-3.2)
+  associated with the producer, providing an internet-wide unique identifier.
+  If not, it is unique within some application-specific scope.
+  The details of a source URI are determined by the producer, the consumer
+  SHOULD treat it as an opaque identifier.
```

Comment:
I recommend that you try to stick with the minimal changes needed to achieve eventIDs uniqueness, and not change other things without a clear need.

The last sentence is unnecessary, and - in my opinion - wrong.

Of course a consumer CAN treat the source as a URI, and not as an opaque identifier!
E.g. I can decide that I'm interested in `/cloudevents/spec/*`, but not in `/cloudevents/website/*`, or that I'm interested in `/cloudevents/*/pull/*`, but not in `/cloudevents/*/issue/*`.

From the perspective of deduplication, it is treated as an opaque identifier, but deduplication isn't the only aspect of the spec.

Reply:
+1 restoring original comments.